Source/Core/Core/Core.cpp

```diff
@@ -335,6 +339,12 @@ static void CpuThread(const std::optional<std::string>& savestate_path, bool del
   // This needs to be delayed until after the video backend is ready.
   DolphinAnalytics::Instance().ReportGameStart();
 
+#ifdef ANDROID
+  // For some reason, calling the JNI function AttachCurrentThread from the CPU thread after a
+  // certain point causes a crash if fastmem is enabled. Let's call it early to avoid that problem.
+  IDCache::GetEnvForThread();
```

Comment:
What do you think of adding a function called AttachCurrentThread that just calls GetEnvForThread, but makes it the purpose/effect of this call more explicit?

Reply:
It seems a bit odd to add a function that just calls another function with no extra processing, and I don't think creating a function named `AttachCurrentThread` is a great idea since there already is a JNI function called `AttachCurrentThread` (which in fact is freestanding if you use the C API). Not that it'll lead to compile errors, but it might be confusing for someone who isn't familiar with AttachCurrentThread and searches for the name in order to look it up. Given that it would only be used for this one workaround, I don't think it makes sense.

Reply:
Hmm, it just looks a bit weird to me to call a "get [something]" function and then proceed to ignore the return value. I feel like the comment expects the reader to know that GetEnvForThread internally calls AttachCurrentThread, but someone who isn't extremely familiar with the JNI code would expect to see a call to "it" (AttachCurrentThread), not GetEnvForThread.

Though now that I know the JNI has a free function called `AttachCurrentThread`, I agree that it wouldn't be a good idea to use that name again. Maybe just do a cast to make it abundantly clear that GetEnvForThread is being called for its side effect (`static_cast<void>(IDCache::GetEnvForThread());`)?

Reply:
Cast added.